### DISCUSSION TITLE
relecture avec phidgets hub0001 VINT

### DISCUSSION THREAD

**Discussion by lartisanes** [artisan-roaster-scope/artisan#1187] [Q&A]
Hello, I have a roaster not referenced in Artisan (15kgs). I bought the phidgets 2x RTD kit and it works great! I would now like to control my roaster with artisan and do curve proofreading. So I need to connect the hub0001 to my console in rj45 (I have the IP address of my roaster). I can't find a direct cable, do I have to go through a vint hub 5000? The French manufacturer of this roaster is not in the artisan database yet (I've been asking him for 6 years :(  ). Can someone help me? Thank you for your answers.

**Comment by bigl00z3** [comment]
To control the roaster is a bit more complicated than just reading probes. You will first need to understand if the roaster can be controlled or not by an external software.

You will need to find out and explain more about the roaster. 
Is it currently possible to remote control it from an external computer (e.g. manufacturer's software). From your post I imagine it is as you mention that there is a RJ45 port.

In most cases, you can control it from artisan if the remote control protocol exists, is accessible (no locked by the manufacturer) and is in the protocols supported by artisan. The list of supported devices and protocols is listed on the [Artisan website](https://artisan-scope.org/devices/index). It could be for example MODBUS, Siemens S7...

Once you find out which "language" your roaster speaks, you can try to speak with it from Artisan. However, if the manufacturer doesn't support Artisan,  it can be quite complicated if they are not willing to provide you with the registers (the addresses) for each command.

Do you have documentation of more details on how your roaster communicates or what type of automate or equipment is in it to control it?

Nico

**Comment by lartisanes** [comment]
thanks for this explication, i understand what you explain and I find all
of the information and pictures (more easy).

I have documentation, i see that.
Thanks a lot !

David

Le dim. 28 mai 2023 à 18:18, Nico Bigler ***@***.***> a
écrit :

> To control the roaster is a bit more complicated than just reading probes.
> You will first need to understand if the roaster can be controlled or not
> by an external software.
>
> You will need to find out and explain more about the roaster.
> Is it currently possible to remote control it from an external computer
> (e.g. manufacturer's software). From your post I imagine it is as you
> mention that there is a RJ45 port.
>
> In most cases, you can control it from artisan if the remote control
> protocol exists, is accessible (no locked by the manufacturer) and is in
> the protocols supported by artisan. The list of supported devices and
> protocols is listed on the Artisan website
> <https://artisan-scope.org/devices/index>. It could be for example
> MODBUS, Siemens S7...
>
> Once you find out which "language" your roaster speaks, you can try to
> speak with it from Artisan. However, if the manufacturer doesn't support
> Artisan, it can be quite complicated if they are not willing to provide you
> with the registers (the addresses) for each command.
>
> Do you have documentation of more details on how your roaster communicates
> or what type of automate or equipment is in it to control it?
>
> Nico
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/artisan-roaster-scope/artisan/discussions/1187#discussioncomment-6023609>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMUYHFDY5LRQR2TVO6EYFBLXIN3D3ANCNFSM6AAAAAAYR2ZLHE>
> .
> You are receiving this because you authored the thread.Message ID:
> <artisan-roaster-scope/artisan/repo-discussions/1187/comments/6023609@
> github.com>
>


-- 
L'Artisanes, Thés, Cafés <http://lartisanes.fr>
Facebook <https://www.facebook.com/lartisanes>
Instagram <https://www.instagram.com/lartisanes/?hl=fr>
Twitter <https://twitter.com/lartisanes>
6, rue Emile Zola     02100 Saint-Quentin
 09.81.39.24.27

**Comment by lartisanes** [comment]
Hi, i find it of information, but i don't know why, the IP adress is always
changing.
On the documentation i don't have more 😞

**Comment by lartisanes** [comment]
![IMG_20230531_151817](https://github.com/artisan-roaster-scope/artisan/assets/53052308/c00708fc-5097-458f-880a-537986b00911)
![IMG_20230530_104222](https://github.com/artisan-roaster-scope/artisan/assets/53052308/682bd0d8-ffcb-4d87-a9ef-50f7e0bae993)
![IMG_20230530_104248](https://github.com/artisan-roaster-scope/artisan/assets/53052308/cbfab296-23cd-470d-a195-88a270889e7f)
![received_588985593098206](https://github.com/artisan-roaster-scope/artisan/assets/53052308/5816b983-c3c9-4201-84c6-958e516fcdc1)

**Reply by poundy** [reply]
your vendor needs to help you understand how to connect to the roaster, if that's what you are trying to achieve. 

From a phidgets perspective, I don't have any experience with them, but there are many resources available on the internet to help you understand how to connect things. But you haven't given us a clear idea on what you want you're attempting to connect to what, so I suspect that anyone who might be able to guide your architecture of your phidgets isn't sure how to help. Your first post was about automating control, not about the value reading that you've already got working - and really, only your vendor can help there. I'm sure many people here can contribute (as Nico already has) but the path of least resistance is getting the vendor to tell you how to interface to their control stack. Otherwise you are going to be guessing (and nobody here has your roaster so there's no point us guessing)

**Reply by bigl00z3** [reply]
I agree with poundy, I don't think many people on this forum will be able to help you much with the IP address setting of the roaster. 
I also agree with his comment that it is not totally clear what you want to achieve. I guess yo uhave to main things:

- **Roaster connection to Artisan via RJ45:**

This is your first step and for this, you will either need to ask the suplier on how to do it or you can also try to look up for the documentation of the automat itself (but based on your pictures I cannot see the brand/model of the control panel). In the documentation you may find usefull information on how to set a static IP address. If it is constantly changing (I imagine everytime you start it) it may be that it it set as DHCP and not static.

- **Using the Phidget HUB0001 to read the additional RTD probes** 

(via Phidget TMP1200) and not via the roaster control panel/RJ45
I am not sure what you mean in your first question when you state :

> I need to connect the hub0001 to my console in rj45

Have you managed to connect your Phidget HUB0001 to your computer and Artisan? 
You would not connect it directly to the roaster RJ45 as it has a USB output and not a RJ45. 
You should be able to read the RTD probes in Artisan via the Phidgets even when not connected to the roaster.

You will first need to solve those 2 **different** communications issues (Phidgets com and RJ45 com). 
Only when this is working fine you can start the complicated and not very straight forward work to try to have the roaster repeat the curves. Do you already know how you want to proceed with this? Using the PID? Or just replicating burner changes from a previous batch? 

Nico

**Reply by lartisanes** [reply]
actuellement, j'utilise phidget TMP1200 en sondes supplémentaires pour une lecture avec artisan.
Connecter artisan au roaster, c'est pour piloter le roaster, mais aussi reproduire en automatique.
je roast toujours en manuel, mais 15t de café, certains je voudrais les faire en automate ( et le programme fourni n'est pas pratique pour ca). J'avais pensé à connecter le phidget (via un hub 5000 rj45). Ou le pc directement en rj 45 sur la connection I/O du roaster.
Pour la marque de l'automate c'est un unitronics v200-18-e6b.
cela fais 6 ans que je demande au fabricant de référencer son appareil chez artisan :)
Pour finir, l'automate est bien en dhcp les aresses ip change tout le temps.
![IMG_20230601_170834](https://github.com/artisan-roaster-scope/artisan/assets/53052308/1f8c0829-6bd5-4900-b12b-bdd9ceb48d18)
![IMG_20230601_164835](https://github.com/artisan-roaster-scope/artisan/assets/53052308/a803f469-7416-4e3f-b999-63241fb85929)
![IMG_20230601_164826](https://github.com/artisan-roaster-scope/artisan/assets/53052308/0e8289dc-ec4d-4255-9e4e-f575c501b83c)

**Reply by poundy** [reply]
you are still shooting in the dark without the vendor telling you how to connect to the PLC in the roaster. You really will need more detail to try to achieve connection to any product not just Artisan. As Nico said, you really need to understand the PLC and whether it's address is fixed or not, or if DHCP. If you can track what the different IP addresses it comes up with, we might be able to GUESS what it's doing but that's just our external guess, much better to have documents we can rely on. I'm happy to help you understand parts of the doco that you find, as are many people here, but am nowhere near your machine and have no idea what parts I'd be looking for. If you do the legwork and find doco, and ask specific questions from that, we might have more ability to assist.  Plus, language being language, things may be somewhat hit and miss as we rely on translations to/from each other...
